### PR TITLE
2659: Don't load other cities when fixed city is set

### PR DESCRIPTION
**Pull request by sarahsporck**
### Short description

Currently it is possible to open IG cities in Aschaffenburg, when directly navigating to the url 
-> https://halloaschaffenburg.de/kreis-bergstrasse/de
This should be avoided, by not loading any other city than the fixed city.

### Proposed changes

- Add a fixedCityContentSwitcher, which checks whether the params `cityCode` is the `fixedCity`

### Resolved issues

Fixes: #2659

### CODE REVIEW

**Review by LeandraH — APPROVED** [review]
Very nice, thanks! Tested in Firefox. Maybe also add a tiny test?

**Review by ztefanie — APPROVED** [review]

**Review by steffenkleinle — APPROVED** [review]

**Review by steffenkleinle — COMMENTED** [review]
Why did you choose to add another component instead of just adding anther condition to the if-clause in https://github.com/digitalfabrik/integreat-app/blob/ccd4c73d658f5ace835bfeca0ec92da028698825/web/src/CityContentSwitcher.tsx#L60?
``` ts
  const { fixedCity } = buildConfig().featureFlags
  if ((!city && !loading) || (fixedCity && cityCode !== fixedCity)) {
```

**Comment by sarahsporck** [comment]
> Why did you choose to add another component instead of just adding anther condition to the if-clause [...]

This way we don't even have to make the request to the endpoint. Therefore, it is faster and uses less traffic.

File: native/src/components/PoiDetails.tsx [hunk 1]
```diff
@@ -49,7 +49,6 @@ type PoiDetailsProps = {
 
 const PoiDetails = ({ poi, language, distance }: PoiDetailsProps): ReactElement => {
   const { t } = useTranslation('pois')
-  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (sarahsporck) [inline native/src/components/PoiDetails.tsx:52]: appears to be unused and threw a linting warning

**Pull request merged**